### PR TITLE
fix(calendar): watch changes to date range

### LINKED ISSUE #10500: md-calendar dynamically change min & max date
I am using 2 md-calendar (not md-datepicker) on same page. On change on first md-datepicker mix & max date will be updated for other md-datepicker. But it is not happening.

**Code Pen for the same is:-**
[](http://codepen.io/lashisu/pen/JWOPex)

I have observed that once START DATE selected and tab moved to END DATE, click on month selector and back to date view, calendar get updated with new min/max date.

**Can you please help me to get rid of this issue?**

### PR DESCRIPTION
**Pull request by mvdiener**
Added a watcher for min/max date changes so md-calendar will update without having to close and reopen.

Fixes #10500.

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
@mvdiener - what is the use case where this fix is actually needed?

**Comment by ThomasBurleson** [comment]
Closing as edge case.

**Comment by mvdiener** [comment]
@ThomasBurleson 
The use case is having two md-calendars (for start and end dates), and having the end date calendar restrict date options based on the selection of the start date calendar. This does not work, unless you close and reopen the calendars. The pull request adds a bit of code to update them live, so they function more like the md-datepicker.

**Comment by ThomasBurleson** [comment]
@mvdiener - thx for the response. 
@EladBezalel - plz review.

**Comment by mvdiener** [comment]
I see this pull request and a few others are all failing due to mdChips issues. Is there a particular branch I should pull to get that working again?

File: src/components/datepicker/js/calendarMonthBody.js [hunk 1]
```diff
@@ -38,6 +42,12 @@
             monthBodyCtrl.generateContent();
           }
         });
+        
+        scope.$watch(function () { return calendarCtrl.minDate + calendarCtrl.maxDate; }, function (range, oldRange) {
```
Comment (EladBezalel) [inline src/components/datepicker/js/calendarMonthBody.js]: Can we avoid this watcher? maybe when `calendarCtrl.minDate` or `calendarCtrl.maxDate` are being changed we should than activate `monthBodyCtrl.generateContent();` ?

Comment (Splaktar) [reply src/components/datepicker/js/calendarMonthBody.js]: @mvdiener is this something that you can address?

**Comment by Splaktar** [comment]
Please follow the [commit message guidelines](https://github.com/angular/material/blob/master/.github/CONTRIBUTING.md#-commit-message-format) in the future.

File: src/components/datepicker/js/calendarMonthBody.js (hunk 1 above)
Comment (mvdiener) [reply src/components/datepicker/js/calendarMonthBody.js]: Sure I will take a look and see if I can figure that out. I've also noticed the yearBodyCtrl will need something similar.

**Comment by Splaktar** [comment]
Moving this to the backlog for now as it's been waiting for review feedback to be addressed for multiple months.

**Comment by Splaktar** [comment]
Closing as review feedback hasn't been addressed.

**Pull request closed without merging**